### PR TITLE
[WIP] Updated Class SoftDeletableEndpointMixin for not get and set the field Deleted with POST/PUT/PATCH.

### PR DESCRIPTION
**Pull request by tiagovizoto**

### CODE REVIEW

File: powerlibs/django/restless/contrib/endpoints/base.py [hunk 1]
```diff
@@ -86,3 +86,33 @@ class SoftDeletableListEndpointMixin:
...
+
+        for field_name in ('deleted', 'updated_at', 'created_at'):
+            if field_name in request.data:
+                return Http400({"error": "Invalid field name: {}".format(field_name)})
+
+        return super().put(request, *args, **kwargs)
+
+    def patch(self, request, *args, **kwargs):
+
+        for field_name in ('deleted', 'created_by', 'updated_at', 'created_at'):
```
Comment (cleberzavadniak) [inline powerlibs/django/restless/contrib/endpoints/base.py]: Why is there 4 fields here and only 3 on line 106?

File: powerlibs/django/restless/contrib/endpoints/tool.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+class SieveListFieldDeleted:
+
+    @staticmethod
+    def delete_field_list(data):
+        if data is not None:
```
Comment (cleberzavadniak) [inline powerlibs/django/restless/contrib/endpoints/tool.py]: I believe you could reimplement the `serialize` method or something like that. I recommend you to research that in `powerlibs-django-restless`, file `modelviews.py` (if I remember right).

File: powerlibs/django/restless/contrib/endpoints/base.py (hunk 1 above)
Comment (tiagovizoto) [reply powerlibs/django/restless/contrib/endpoints/base.py]: Coloquei tudo junto. 

``` Não permitir que atualize-se (via put ou patch) campos como `created_by`, `created_at` e `updated_at`.```

e 

```SoftDeletableEndpointMixin, sumir com o campo deleted no GET e impedir que seja setado via POST/PUT/PATCH.```

File: powerlibs/django/restless/contrib/endpoints/base.py [hunk 3]
```diff
@@ -86,3 +86,33 @@ class SoftDeletableListEndpointMixin:
     def get_query_set(self, request, *args, **kwargs):
         queryset = super().get_query_set(request, *args, **kwargs)
         return queryset.filter(deleted=False)
+
+
+class SieveListEndpoint:
```
Comment (cleberzavadniak) [inline powerlibs/django/restless/contrib/endpoints/base.py]: I don't think you should create a class for that. The behaviour of getting rid of `deleted` field should be implemented on `SoftDeletableEndpoint`. The same for `created_by` on `OwnedEndpoint`.

File: powerlibs/django/restless/contrib/endpoints/base.py [hunk 4]
```diff
@@ -71,6 +71,40 @@ def get_query_set(self, request, *args, **kwargs):
 
 
 class SoftDeletableDetailEndpointMixin:
+    def serialize(self, data):
+        data = super().serialize(data)
+        if 'deleted' in data:
+            del data['deleted']
```
Comment (cleberzavadniak) [inline powerlibs/django/restless/contrib/endpoints/base.py]: Nope.
There is already a specific way of filtering fields.

See:

https://github.com/DroneMapp/powerlibs-django-restless/blob/master/powerlibs/django/restless/modelviews.py#L69

https://github.com/DroneMapp/powerlibs-django-restless/blob/master/powerlibs/django/restless/modelviews.py#L6

https://github.com/DroneMapp/powerlibs-django-restless/blob/master/powerlibs/django/restless/models.py#L77

File: powerlibs/django/restless/contrib/endpoints/base.py [hunk 5]
```diff
@@ -71,6 +71,40 @@ def get_query_set(self, request, *args, **kwargs):
...
+
+        return data
+
+    def put(self, request, *args, **kwargs):
```
Comment (cleberzavadniak) [inline powerlibs/django/restless/contrib/endpoints/base.py]: 1- `created_by` and other fields have **nothing** to do with this mixin.
2- Reimplementing almost entirely the `put` method. It seems too agressive.
3- Two calls to the "parent" method (with `super`) on two different returns. It's a bad smell.
4- Why would you fill `result` (line 83) if it is not even used to check for invalid fields?

So, please, take a deep breath, think about what you want to do, and try again.

**Pull request closed without merging**